src/hooks/useFilters.ts

```diff
@@ -0,0 +1,94 @@
+import { ServiceRequestSummary } from '@types';
+import { FilterMatchMode } from 'primereact/api';
+import { DataTableFilterMeta } from 'primereact/datatable';
+import { useMemo, useState } from 'react';
+
+export function passThroughFilter() { return true; }
+
+export const defaultExternalFilters = {
+  global_urgent: null,
+  owner_and_pet: '',
+  global_species: [],
+};
+export const defaultInternalFilters: DataTableFilterMeta = {
+  global: { value: null, matchMode: FilterMatchMode.CONTAINS },
+  service_category: { value: null, matchMode: FilterMatchMode.IN },
+  'team_member.email': { value: null, matchMode: FilterMatchMode.IN },
+  created_at: { value: null, matchMode: FilterMatchMode.CUSTOM },
+  'client.first_name': { value: '', matchMode: FilterMatchMode.CUSTOM },
+};
+export default function useFilters(items: ServiceRequestSummary[]) {
+  const [externalFilters, setExternalFilters] = useState(defaultExternalFilters);
+  const [internalFilters, setInternalFilters] = useState(defaultInternalFilters);
+
+  const filterByUrgent = (request: ServiceRequestSummary) => {
+    if (externalFilters.global_urgent) {
+      return request.urgent === externalFilters.global_urgent;
+    }
+    return passThroughFilter();
+  };
+  const filterBySpecies = (request: ServiceRequestSummary) => {
+    if (externalFilters.global_species.length > 0) {
+      return externalFilters.global_species.some(
+        (currentSpeciesFilters) => currentSpeciesFilters.includes(request.pet.species),
+      );
+    }
+    return passThroughFilter();
+  };
+  const filterByName = (request: ServiceRequestSummary) => {
+    const filterValue = externalFilters.owner_and_pet;
+    if (filterValue) {
+      const filteredFieldsContent = [
+        request.client.first_name,
+        request.client.last_name,
+        request.pet.name,
+      ];
+      return filteredFieldsContent.some(
```

Comment:
If the user enters the first and last name into the filter it doesn't seem to work. I think you just need to join the first name and last name search